### DISCUSSION TITLE
How to get `$ref`s working?

### DISCUSSION THREAD

**Discussion by OnkelTem** [redhat-developer/vscode-yaml#772] [General]
I read the documentation but couldn't find anything about using `$ref`s and how they get resolved.

Consider the following definitions:

**foo.schema.json**
```json
{
  "$id": "https://someprefix/foo",
  "title": "Foo",
  "type": "object",
  "properties": {
    "bar": { "$ref": "bar#" }
  },
  "required": ["bar"]
}
```

**bar.schema.json**
```json
{
  "$id": "https://someprefix/bar",
  "title": "Bar",
  "type": "object",
  "properties": {
    "value": { "type": "string" }
  },
  "required": ["value"]
}
```

As you see the `https://someprefix/foo` schema depends on the `https://someprefix/bar` one. The reference to from `foo` to `bar` is performed via `{ "$ref": "bar#" }`. Unfortunately it doesn't work with this plugin:

```yaml
# yaml-language-server: $schema=./foo.schema.json
bar:
  value: "hello"
```
results to a validation error: 
```
Problems loading reference 'file:///projects/bar': Unable to load schema from '/projects/bar': No content.YAML(768)
```

Is there a way to load all the schemas from a directory to get this resolution working? This is how [`Ajv` ](https://ajv.js.org/) resolution works for example.

**Comment by ssbarnea** [comment]
Read the docs from https://json-schema.org/understanding-json-schema/structuring.html#id19 and be very careful about the relative URLs. 

Hint: `#/$defs/foo` is what you want, and putting the ref inside `$defs`

**Reply by derberg** [reply]
@ssbarnea Hey, I'm having similar issue, using JSON Schema 7

My schema is: https://github.com/asyncapi/spec-json-schemas/blob/master/schemas/2.4.0.json

I have this https://github.com/asyncapi/spec-json-schemas/blob/master/schemas/2.4.0.json#L601 and this reference is not resolved, even though the `$id` points to [`"$id": "http://json-schema.org/draft-07/schema"`](https://github.com/asyncapi/spec-json-schemas/blob/master/schemas/2.4.0.json#L483). So references should be solved without issues with http://json-schema.org/draft-07/schema#/definitions/schemaArray.

Is base URI actually supported? https://github.com/redhat-developer/vscode-yaml/issues/773

![Screenshot 2022-06-27 at 17 04 29](https://user-images.githubusercontent.com/6995927/175994617-070c4311-0825-4ff7-a83b-3a4ebf82110f.png)

I see this plugin has custom resolver https://github.com/codeclown/yaml-language-server/blob/main/src/languageservice/services/yamlSchemaService.ts and I'm having hard time figuring out all the details
